pkg/services/sqlstore/user.go

```diff
@@ -620,6 +621,21 @@ func (ss *SQLStore) SearchUsers(ctx context.Context, query *models.SearchUsersQu
 			whereParams = append(whereParams, query.OrgId)
 		}
 
+		// user only sees the users in which it has user read access too
+		if ss.Cfg.IsFeatureToggleEnabled(featuremgmt.FlagAccesscontrol) && query.SignedInUser != nil {
```

Comment:
question: should we return an error or empty result if `SignedInUser` isn't provided? It's quite easy to forget to set this field, which could then result in us listing more users than the caller has access to.

Reply:
then we can check that it needs to be set.